src/platform/packages/shared/kbn-unified-metrics-grid/src/common/utils/esql/create_esql_query.test.ts

```diff
@@ -102,7 +108,10 @@ TS metrics-*
   it('should cast numeric fields in multiple dimensions', () => {
     const query = createESQLQuery({
       metric: mockMetric,
-      dimensions: ['cpu.cores', 'host.name'],
+      dimensions: [
```

Comment:
This test is generating `cpu.cores::STRING IS NOT NULL`. Do we need to cast the field to check whether it's not null?